alea/runner.py

```diff
@@ -0,0 +1,400 @@
+
+    def __init__(
+            self,
+            statistical_model: str,
+            statistical_model_args: dict = None,
+            parameter_definition: Optional[dict or list] = None,
+            confidence_level: float = 0.9,
+            confidence_interval_kind: str = 'central',
+            confidence_interval_threshold: Callable[[float], float] = None,
+            poi: str = None,
+            hypotheses: list = None,
+            common_generate_values: dict = None,
+            true_generate_values: dict = None,
+            n_mc: int = 1,
+            toydata_file: str = None,
+            toydata_mode: str = None,
+            metadata: dict = None,
+            output_file: str = 'test_toymc.hdf5',
+        ):
+        """
+        Initialize statistical model,
+        parameters list, and generate values list
+        """
+        statistical_model_class = locate(statistical_model)
+        if statistical_model_class is None:
+            raise ValueError(f'Could not find {statistical_model}!')
+        if not inspect.isclass(statistical_model_class):
+            raise ValueError(f'{statistical_model_class} is not a class!')
+        if not issubclass(statistical_model_class, StatisticalModel):
+            raise ValueError(f'{statistical_model_class} is not a subclass of StatisticalModel!')
+        self.statistical_model = statistical_model_class(
+            parameter_definition=parameter_definition,
+            confidence_level=confidence_level,
+            confidence_interval_kind=confidence_interval_kind,
+            confidence_interval_threshold=confidence_interval_threshold,
+            **(statistical_model_args if statistical_model_args else {}),
+        )
+
+        self.poi = poi
+        self.hypotheses = hypotheses if hypotheses else []
+        self.common_generate_values = common_generate_values
+        self.true_generate_values = true_generate_values
+        self.n_mc = n_mc
+        self.toydata_file = toydata_file
+        self.toydata_mode = toydata_mode
+        self.metadata = metadata if metadata else {}
+        self.output_file = output_file
+
+        self.parameter_list, self.result_list, self.result_dtype = self._get_parameter_list()
+
+        self.generate_values = self._get_generate_values()
+
+    def _get_parameter_list(self):
+        """Get parameter list and result list from statistical model"""
+        parameter_list = sorted(self.statistical_model.get_parameter_list())
+        result_list = parameter_list + ['ll', 'dl', 'ul']
+        result_dtype = [(n, float) for n in parameter_list]
+        result_dtype += [(n, float) for n in ['ll', 'dl', 'ul']]
+        # try:
+        #     parameter_list += self.statistical_model.additional_parameters
+        #     result_dtype += [(n, float) for n in self.statistical_model.additional_parameters]
+        # except:
+        #     pass
+        return parameter_list, result_list, result_dtype
+
+    def _get_generate_values(self):
```

Comment:
The naming inside here mixed hypothesis and generate values even for when hypothesis is not "true"-- I think this is unfortunate, separating them conceptually is a key to getting the concepts right in my mind.

Reply:
updated in previous commits.